### PR TITLE
Upgrade Kotlin to 1.4.10, coroutines 1.4.1.

### PR DESCRIPTION
**Pull request by zach-klippenstein**

### CODE REVIEW

File: versions.properties [hunk 1]
```diff
@@ -32,16 +32,16 @@ version.io.mockk..mockk=1.10.0
...
+version.kotlinx.coroutines=1.4.1
+version.kotlinx.serialization=1.0.0
 version.me.champeau.gradle..jmh-gradle-plugin=0.5.1
 version.moshi=1.9.3
 version.okio=2.7.0
 version.org.hamcrest..hamcrest-core=2.2
 version.org.jetbrains..annotations=19.0.0
 version.org.jetbrains.dokka..dokka-gradle-plugin=0.10.1
-version.org.jetbrains.kotlin..kotlin-reflect=1.3.72
+version.org.jetbrains.kotlin..kotlin-reflect=1.4.0
```
Comment (0legg) [inline versions.properties:44]: Should it be 1.4.10 as well?

Comment (zach-klippenstein) [reply versions.properties]: Yep, fixed.

File: versions.properties [hunk 2]
```diff
@@ -32,16 +32,16 @@ version.io.mockk..mockk=1.10.0
 version.io.reactivex.rxjava2..rxandroid=2.1.1
 version.io.reactivex.rxjava2..rxjava=2.2.19
 version.junit.junit=4.13
-version.kotlin=1.3.72
-version.kotlinx.coroutines=1.3.7
-version.kotlinx.serialization=0.20.0
+version.kotlin=1.4.10
+version.kotlinx.coroutines=1.4.1
+version.kotlinx.serialization=1.0.0
```
Comment (0legg) [inline versions.properties:37]: 1.0.1 was released 3 weeks ago with few patches

File: workflow-testing/src/main/java/com/squareup/workflow1/testing/WorkflowTestRuntime.kt [hunk 3]
```diff
@@ -435,9 +435,9 @@ private fun <T> unwrapCancellationCause(block: () -> T): T {
   } catch (e: CancellationException) {
     throw generateSequence(e as Throwable) { e.cause }
         // Stop the sequence if an exception's cause is itself.
-        .scanReduce { error, cause ->
+        .runningReduce { error, cause ->
```
Comment (0legg) [inline workflow-testing/src/main/java/com/squareup/workflow1/testing/WorkflowTestRuntime.kt:438]: Nit: for the same behavior you can utilize `windowed` function, and it'll be more straightforward for a reader. I won't  guarantee it'll fit well with what you're trying to achieve in the next call, but anyway this part could benefit from some rethink of conditions and tinkering with stdlib functions.

At the same time, no objections in merging it as is.

**Review by 0legg — APPROVED** [review]

File: workflow-testing/src/main/java/com/squareup/workflow1/testing/WorkflowTestRuntime.kt (hunk 3 above)
Comment (zach-klippenstein) [reply workflow-testing/src/main/java/com/squareup/workflow1/testing/WorkflowTestRuntime.kt:438]: I don't see `windowed` in the docs – what am I missing?

Comment (zach-klippenstein) [reply workflow-testing/src/main/java/com/squareup/workflow1/testing/WorkflowTestRuntime.kt:438]: Yea, we'd need another operator to reduce the emitted lists I believe.

File: versions.properties (hunk 2 above)
Comment (zach-klippenstein) [reply versions.properties:37]: This is only used for samples, but better to be up-to-date.

File: workflow-testing/src/main/java/com/squareup/workflow1/testing/WorkflowTestRuntime.kt (hunk 3 above)
Comment (0legg) [reply workflow-testing/src/main/java/com/squareup/workflow1/testing/WorkflowTestRuntime.kt:438]: Either https://kotlinlang.org/api/latest/jvm/stdlib/kotlin.collections/zip-with-next.html or https://kotlinlang.org/api/latest/jvm/stdlib/kotlin.collections/windowed.html could be a good candidate, but I don't know how will it help with the methods down the chain.

Comment (zach-klippenstein) [reply workflow-testing/src/main/java/com/squareup/workflow1/testing/WorkflowTestRuntime.kt:438]: Yea, both those would require another operator to flatten the lists they return. `runningReduce` does it in one. I'm not sure which is more readable.

**Review by rjrjr — APPROVED** [review]

**Pull request merged**